`@PostCreate` does  throw an Exception

See the reproduction in the POC here https://github.com/EugenMayer/java-spring-pocs/blob/master/src/test/java/de/kontextwork/poc/spring/blaze/user/BlazeServiceTest.java#L52

With the `View` at https://github.com/EugenMayer/java-spring-pocs/blob/master/src/main/java/de/kontextwork/poc/spring/blaze/user/model/domain/BlazeUserCreateView.java#L11

```
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.blazebit.persistence.view.impl.EntityViewManagerImpl.create(EntityViewManagerImpl.java:306)
	... 91 more
Caused by: java.lang.IncompatibleClassChangeError
	at de.kontextwork.dwcore.user.model.domain.UserCreateView_$$_javassist_entityview_.<init>(UserCreateView_$$_javassist_entityview_.java)
	... 96 more
```

BP: master HEAD
Java: 11
Hibernat: 5.4.5
spring boot: 2.1.9

--- a/build.sh
+++ b/build.sh
@@ -27,7 +27,7 @@ mvn -version
 PROPERTIES="$PROPERTIES -Duser.country=US -Duser.language=en"
 
 if [ "$BUILD_JDK" != "" ]; then
-  PROPERTIES="$PROPERTIES -Djava.version=$BUILD_JDK"
+  PROPERTIES="$PROPERTIES -Djava.version=$BUILD_JDK -Dtest.java.version=$BUILD_JDK"
 fi
 
 if [ "$JDK" != "" ]; then

--- a/entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/proxy/ProxyFactory.java
+++ b/entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/proxy/ProxyFactory.java
@@ -2257,7 +2257,8 @@ public class ProxyFactory {
             } else {
                 postCreateMethodDescriptor = "()V";
             }
-            bc.addInvokespecial(managedViewType.getJavaType().getName(), postCreateMethod.getName(), postCreateMethodDescriptor);
+            // Since we implement this default method, we must use invokevirtual
+            bc.addInvokevirtual(cc, postCreateMethod.getName(), postCreateMethodDescriptor);
             bc.addReturn(null);
 
             CodeAttribute newCodeAttribute = bc.toCodeAttribute();

--- a/entity-view/testsuite/pom.xml
+++ b/entity-view/testsuite/pom.xml
@@ -31,6 +31,9 @@
     <name>Blazebit Persistence Entity-View Testsuite</name>
 
     <properties>
+        <!-- We use Java 8 for a few tests -->
+        <test.java.version>1.8</test.java.version>
+        <java.version>${test.java.version}</java.version>
         <module.name>com.blazebit.persistence.view.testsuite</module.name>
     </properties>
 
@@ -199,7 +202,7 @@
                             <systemPropertyVariables>
                                 <jdbc.url>jdbc:mysql://localhost:3306/test?useUnicode=true&amp;characterEncoding=utf8</jdbc.url>
                                 <jdbc.user>root</jdbc.user>
-                                <jdbc.password />
+                                <jdbc.password/>
                                 <jdbc.driver>com.mysql.jdbc.Driver</jdbc.driver>
                             </systemPropertyVariables>
                         </configuration>
@@ -233,7 +236,7 @@
                             <systemPropertyVariables>
                                 <jdbc.url>jdbc:mysql://localhost:3306/test?useUnicode=true&amp;characterEncoding=utf8</jdbc.url>
                                 <jdbc.user>root</jdbc.user>
-                                <jdbc.password />
+                                <jdbc.password/>
                                 <jdbc.driver>com.mysql.jdbc.Driver</jdbc.driver>
                             </systemPropertyVariables>
                         </configuration>
@@ -300,8 +303,8 @@
                             <excludedGroups>com.blazebit.persistence.testsuite.base.jpa.category.NoSQLite,${jpa.excludedGroups}</excludedGroups>
                             <systemPropertyVariables>
                                 <jdbc.url>jdbc:sqlite:test.db</jdbc.url>
-                                <jdbc.user />
-                                <jdbc.password />
+                                <jdbc.user/>
+                                <jdbc.password/>
                                 <jdbc.driver>org.sqlite.JDBC</jdbc.driver>
                             </systemPropertyVariables>
                         </configuration>

--- a/entity-view/testsuite/src/test/java/com/blazebit/persistence/view/testsuite/proxy/ProxyFactoryJava8Test.java
+++ b/entity-view/testsuite/src/test/java/com/blazebit/persistence/view/testsuite/proxy/ProxyFactoryJava8Test.java
@@ -1,0 +1,69 @@
+/*
+ * Copyright 2014 - 2019 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.view.testsuite.proxy;
+
+import com.blazebit.persistence.spi.PackageOpener;
+import com.blazebit.persistence.testsuite.entity.Person;
+import com.blazebit.persistence.view.impl.metamodel.ManagedViewTypeImplementor;
+import com.blazebit.persistence.view.impl.proxy.ConstructorReflectionInstantiator;
+import com.blazebit.persistence.view.impl.proxy.ObjectInstantiator;
+import com.blazebit.persistence.view.impl.proxy.ProxyFactory;
+import com.blazebit.persistence.view.metamodel.ViewMetamodel;
+import com.blazebit.persistence.view.metamodel.ViewType;
+import com.blazebit.persistence.view.testsuite.AbstractEntityViewTest;
+import com.blazebit.persistence.view.testsuite.proxy.model.DocumentClassView;
+import com.blazebit.persistence.view.testsuite.proxy.model.DocumentCreateView;
+import com.blazebit.persistence.view.testsuite.proxy.model.DocumentJava8View;
+import com.blazebit.persistence.view.testsuite.proxy.model.UnsafeDocumentClassView;
+import com.blazebit.reflection.ReflectionUtils;
+import org.junit.Test;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+/**
+ *
+ * @author Christian Beikov
+ * @since 1.0.0
+ */
+public class ProxyFactoryJava8Test extends AbstractEntityViewTest {
+
+    private final ProxyFactory proxyFactory = new ProxyFactory(false, PackageOpener.NOOP);
+
+    private ViewMetamodel getViewMetamodel() {
+        return build(
+                DocumentJava8View.class
+        );
+    }
+
+    @Test
+    public void testProxyCreateInitialization() throws Exception {
+        ViewType<DocumentJava8View> viewType = getViewMetamodel().view(DocumentJava8View.class);
+        Class<? extends DocumentJava8View> proxyClass = proxyFactory.getProxy(evm, (ManagedViewTypeImplementor<DocumentJava8View>) viewType, null);
+
+        DocumentJava8View instance = proxyClass.getConstructor().newInstance();
+
+        assertEquals("INIT", instance.getName());
+    }
+}

--- a/entity-view/testsuite/src/test/java/com/blazebit/persistence/view/testsuite/proxy/model/DocumentJava8View.java
+++ b/entity-view/testsuite/src/test/java/com/blazebit/persistence/view/testsuite/proxy/model/DocumentJava8View.java
@@ -1,0 +1,42 @@
+/*
+ * Copyright 2014 - 2019 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.view.testsuite.proxy.model;
+
+import com.blazebit.persistence.testsuite.entity.Document;
+import com.blazebit.persistence.testsuite.entity.Person;
+import com.blazebit.persistence.view.EntityView;
+import com.blazebit.persistence.view.Mapping;
+import com.blazebit.persistence.view.PostCreate;
+import com.blazebit.persistence.view.testsuite.proxy.model.IdHolderView;
+
+/**
+ *
+ * @author Christian Beikov
+ * @since 1.4.0
+ */
+@EntityView(Document.class)
+public interface DocumentJava8View extends IdHolderView<Long> {
+
+    @PostCreate
+    default void init() {
+        setName("INIT");
+    }
+
+    public String getName();
+
+    public void setName(String name);
+}

